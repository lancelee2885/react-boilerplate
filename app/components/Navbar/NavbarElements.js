import styled from 'styled-components';
import { Link as LinkRouter } from 'react-router-dom';

export const Nav = styled.nav`
  background: black;
  opacity: 0.8;
  height: 80px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1rem;
  position: sticky;
  top: 0;
  z-index: 10;
`;

export const NavbarContainer = styled.div`
  display: flex;
  justify-content: center;
  height: 80px;
  z-index: 1;
  width: 100%;
  padding: 0 24px;
  max-width: 1100px;
`;

export const NavMenu = styled.ul`
  display: flex;
  align-items: center;
  list-style: none;
  text-align: center;
  margin-top: 20px;
`;

export const NavItem = styled.li`
  height: auto;
`;

export const NavLinks = styled(LinkRouter)`
  color: #fff;
  justify-self: flex-start;
  cursor: pointer;
  font-size: 1rem;
  display: flex;
  align-items: center;
  margin-left: 12px;
  margin-right: 12px;
  font-weight: bold;
  text-decoration: none;
`;
