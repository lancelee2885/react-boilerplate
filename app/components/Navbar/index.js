import React from 'react';
import { IconContext } from 'react-icons/lib';

import {
  Nav,
  NavbarContainer,
  NavLogo,
  NavMenu,
  NavItem,
  NavLinks,
} from './NavbarElements';

/** Navbar: Navigation bar appears and stays on top of the page throughout scrolling.
 * Props:
 *  - toggle: function rec'd from parent(Home).
 *
 * States:
 *  - scrollNav: When scroll over a certain distance, it will cause a state change.
 *
 * @category React Components
 * @component
 */
function Navbar() {
  return (
    <IconContext.Provider value={{ color: '#fff' }}>
      <Nav>
        <NavbarContainer>
          <NavLogo to="/">LOGO</NavLogo>
          <NavMenu>
            <NavItem>
              <NavLinks to="/cryptos" exact="true">
                Cryptos
              </NavLinks>
            </NavItem>
            <NavItem>
              <NavLinks to="/addcryptos" exact="true">
                Add Your Crypto
              </NavLinks>
            </NavItem>
          </NavMenu>
        </NavbarContainer>
      </Nav>
    </IconContext.Provider>
  );
}

export default Navbar;
