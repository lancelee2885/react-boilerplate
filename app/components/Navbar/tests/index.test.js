import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter } from 'react-router-dom';

import Navbar from '../index';
import configureStore from '../../../configureStore';

describe('<Navbar />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <Provider store={store}>
          <IntlProvider locale="en">
            <BrowserRouter>
              <Navbar />
            </BrowserRouter>
          </IntlProvider>
        </Provider>,
      )
      .toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });
});
