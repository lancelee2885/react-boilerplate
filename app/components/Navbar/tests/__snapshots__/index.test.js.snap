// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render and match the snapshot 1`] = `
<nav
  className="NavbarElements__Nav-sc-1a5xcmq-0 RMoKH"
>
  <div
    className="NavbarElements__NavbarContainer-sc-1a5xcmq-1 jjbpPb"
  >
    <ul
      className="NavbarElements__NavMenu-sc-1a5xcmq-2 dZGELk"
    >
      <li
        className="NavbarElements__NavItem-sc-1a5xcmq-3 eiByiO"
      >
        <a
          className="NavbarElements__NavLinks-sc-1a5xcmq-4 chyWMJ"
          exact="true"
          href="/"
          onClick={[Function]}
        >
          Cryptos
        </a>
      </li>
      <li
        className="NavbarElements__NavItem-sc-1a5xcmq-3 eiByiO"
      >
        <a
          className="NavbarElements__NavLinks-sc-1a5xcmq-4 chyWMJ"
          exact="true"
          href="/add"
          onClick={[Function]}
        >
          Add Crypto
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
